microsoft/testsuites/performance/networkperf.py

```diff
@@ -424,6 +435,568 @@ def perf_sockperf_latency_tcp_synthetic_busy_poll(self, result: TestResult) -> N
+        server.reboot()
+
+        perf_ntttcp(
+            test_result=result,
+            client=client,
+            server=server,
+            server_nic_name=server_nic_name,
+            client_nic_name=client_nic_name,
+            udp_mode=True,
+        )
+
+    def _configure_passthrough_nic_for_node(
+        self,
+        node: Node,
+    ) -> Tuple[RemoteNode, str]:
+        ctx = get_node_context(node)
+        if not ctx.passthrough_devices:
+            raise SkippedException("No passthrough devices found for node")
+
+        # Configure the nw interface on guest
+        node.execute(
+            cmd="dhclient",
+            sudo=True,
+            expected_exit_code=0,
+            expected_exit_code_failure_message="dhclient command failed",
+        )
+
+        lspci = node.tools[Lspci]
+        pci_devices = lspci.get_devices_by_type(
+            constants.DEVICE_TYPE_SRIOV, force_run=True
+        )
+        device_addr = None
+
+        # Get the first non-virtio device
+        for device in pci_devices:
+            kernel_driver = lspci.get_used_module(device.slot)
+            if kernel_driver != "virtio-pci":
+                device_addr = device.slot
+                break
+
+        # Get the interface name
+        err_msg: str = "Can't find interface from PCI address"
+        device_path = node.execute(
+            cmd=(
+                "find /sys/class/net/*/device/subsystem/devices"
+                f" -name '*{device_addr}*'"
+            ),
+            sudo=True,
+            shell=True,
+            expected_exit_code=0,
+            expected_exit_code_failure_message=err_msg,
+        ).stdout
+
+        pattern = re.compile(r"/sys/class/net/(?P<INTERFACE_NAME>\w+)/device")
+        interface_name_raw = find_group_in_lines(
+            pattern=pattern,
+            lines=device_path,
+        )
+        interface_name = interface_name_raw.get("INTERFACE_NAME", "")
+        assert interface_name, "Can not find interface name"
+
+        # Get the interface ip
+        err_msg = f"Failed to get interface details for: {interface_name}"
+        interface_details = node.execute(
+            cmd=f"ip addr show {interface_name}",
+            sudo=True,
+            expected_exit_code=0,
+            expected_exit_code_failure_message=err_msg,
+        ).stdout
+        ip_regex = re.compile(r"\binet (?P<INTERFACE_IP>\d+\.\d+\.\d+\.\d+)/\d+\b")
+        interface_ip = find_group_in_lines(
+            lines=interface_details,
+            pattern=ip_regex,
+            single_line=False,
+        )
+        passthrough_nic_ip = interface_ip.get("INTERFACE_IP", "")
+        assert passthrough_nic_ip, "Can not find interface IP"
+
+        test_node = cast(RemoteNode, node)
+        test_node.internal_address = passthrough_nic_ip
+
+        return test_node, interface_name
+
+    def _get_host_as_server(self, variables: Dict[str, Any]) -> RemoteNode:
+        ip = variables.get("pa_ip", "")
```

Comment:
What means "pa" here? it should provide clear names, so it's easy to use. For host operations, it should be wrap as features, so it doesn't need to handle connection from test cases.

Reply:
@squirrelsc Agree.

This is the host machine used to measure perf test between VM on one physical server and another physical server NIC. I changed now it to `baremetal_host_` now.